slick/src/main/scala/za/co/absa/fadb/slick/SlickPgFunctionWithStatusSupport.scala

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2022 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.fadb.slick
+
+import slick.jdbc.{GetResult, PositionedResult}
+import za.co.absa.fadb.statushandling.FunctionStatus
+
+import scala.util.Try
+
+/**
+  * An extension of the [[SlickPgFunction]] mix-in trait to add support of status handling
+  * This trait expects another mix-in of [[za.co.absa.fadb.statushandling.StatusHandling StatusHandling]] (or implementation of `checkStatus` function)
+  * @tparam T - The input type of the function
+  * @tparam R - The return type of the function
+  */
+trait SlickPgFunctionWithStatusSupport[T, R] extends SlickPgFunction[T, R] {
+
+  /**
+    * Function which should actually check the status code returned by the DB function. Expected to got implemented by
```

Comment:
the `core` and `slick` module are very coupled together in terms of imports and comments. I am a bit unsure about all this. Usually the traits / most abstract classes are not aware of the successors. Usually these just briefly outline what the desired intention should be. And then, next layer, the successor(s) are dealing with those details.

Re the imports, I'm not sure about the structure of the repo. `FunctionStatus` quite belongs into the 'core' -> 'statushandling', but it's also used here.

An idea, why not to 'merge' those 2 modules? Still can be in different package path,like it's now, but maybe under the same top level directory at least?

Reply:
Don't get me wrong, I love the code docs you provided, it's just that I think all these abstract things should not be aware of the successors - this info might be in the successors themselves

Reply:
Slick can know (and should) about the core.
Core should not be referring to Slick in any way. It does expect a particular implementation to fill in the abstract gaps - that's by design.
I don't want to drag in Slick into every user of fa-db. Right now, there's no other option, as it's the only implementation. But that is the reason I don't want to put them into one module.

Reply:
Okay I'll think about it. If you won't hear from me in regards to this issue, then I either agree or don't have a better solution :D